packages/ai-mcp/src/browser/mcp-command-contribution.ts

```diff
@@ -0,0 +1,121 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { AICommandHandlerFactory } from '@theia/ai-core/lib/browser/ai-command-handler-factory';
+import { CommandContribution, CommandRegistry, MessageService } from '@theia/core';
+import { QuickInputService } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { MCPServerManager } from '../common/mcp-server-manager';
+import { ToolInvocationRegistry, ToolRequest } from '@theia/ai-core';
+
+export const StartMCPServer = {
+    id: 'mcp.startserver',
+    label: 'MCP: Start MCP Server',
+};
+export const StopMCPServer = {
+    id: 'mcp.stopserver',
+    label: 'MCP: Stop MCP Server',
+};
+
+@injectable()
+export class MCPCommandContribution implements CommandContribution {
+    @inject(AICommandHandlerFactory)
+    protected readonly commandHandlerFactory: AICommandHandlerFactory;
+
+    @inject(QuickInputService)
+    protected readonly quickInputService: QuickInputService;
+
+    @inject(MessageService)
+    protected messageService: MessageService;
+
+    @inject(MCPServerManager)
+    protected readonly mcpServerManager: MCPServerManager;
+
+    @inject(ToolInvocationRegistry)
+    protected readonly toolInvocationRegistry: ToolInvocationRegistry;
+
+    private async getMCPServerSelection(serverNames: String[]): Promise<string | undefined> {
+        if (!serverNames || serverNames.length === 0) {
```

Comment:
I think this message should be moved out of the function. The `getMCPServerSelection` should not even be called if the array is empty. Instead a more specialized error message should be produced by the caller of this method and then not even invoke it.

As a sanity check I would still return `undefined` here though